src/Main/Main.js

```diff
@@ -13,17 +13,27 @@ export class Main extends Component {
   componentDidMount() {
     this.props.fetchFromFirebase();
     this.props.getUserLocation();
-
   }
 
-  componentWillUpdate(nextProps) {
-    if ( this.props.competitions.length !== nextProps.competitions.length) {
-      const { activeUser, competitions } =  nextProps;
-      loadUserComps(activeUser, competitions);
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.props.liveUser &&
+      (this.props.competitions.length !== nextProps.competitions.length)) {
+      this.props.loadUserComps(this.props.activeUser, nextProps.competitions);
     }
+    return true;
```

Comment:
I ran into a lot of problems trying to dispatch functions from shouldComponentUpdate. I used 'componentWillReceiveProps' instead. You still get next props, but I think it's the lifecycle method best suited for this thing (Mostly because I got PAGES of errors telling me not to use shouldComponentUpdate when I updated the same component's props with it)

Reply:
I'll give that one another go. Thanks!